lib/elixir/lib/module/types/descr.ex

```diff
@@ -95,18 +135,35 @@ defmodule Module.Types.Descr do
   @compile {:inline, intersection: 3}
   defp intersection(:bitmap, v1, v2), do: bitmap_intersection(v1, v2)
   defp intersection(:atom, v1, v2), do: atom_intersection(v1, v2)
+  defp intersection(:dynamic, v1, v2), do: dynamic_intersection(v1, v2)
 
   @doc """
   Computes the difference between two types.
   """
   def difference(left = %{}, right = %{}) do
+    if gradual?(left) or gradual?(right) do
+      {left_dynamic, left_static} = Map.pop(left, :dynamic, left)
+      {right_dynamic, right_static} = Map.pop(right, :dynamic, right)
+      dynamic_part = difference_static(left_dynamic, right_static)
+
+      if empty?(dynamic_part),
+        do: @none,
```

Comment:
Are we sure this is the case? Should it be:

```suggestion
        do: difference_static(left_static, right_dynamic),
```

If I am right, can we have a test that shows this?

Reply:
You are right, but since we have the invariant that `left_dynamic >= left_static` and `right_dynamic >= right_static` the type `left_dynamic \ right_static` is a supertype of `left_static \ right_dynamic`, 
So if it's empty, the second is empty as well and we shouldn't compute it.

Reply:
So basically, I am not right, and the current code is good as is? :D

Reply:
Correct :D